lib/matplotlib/path.py

```diff
@@ -1033,3 +1197,228 @@ def get_path_collection_extents(
+    or np.isclose(theta, np.pi) and np.isclose(phi, np.pi/2):
+        return width/2
+    # to calculate the offset for _joinstyle == 'miter', imagine aligning the
+    # corner so that one line comes in along the negative x-axis, and another
+    # from above, making an angle $\theta$ with the negative x-axis. The tip of
+    # the new corner created by the markeredge stroke will be at the point
+    # where the two outer edge of the markeredge stroke intersect.  in the
+    # orientation described above, the outer edge of the stroke aligned with
+    # the x axis will obviously have equation $y = -w/2$ where $w$ is the
+    # markeredgewidth. WLOG, the stroke coming in from above at an angle
+    # $\theta$ from the negative x-axis will have equation
+    # $$-(\tan(\theta) x + \frac{w}{2\cos(\theta)}.$$
+    # the intersection of these two lines is at $y = w/2$, and we can solve for
+    # $x = \cot(\theta) (\frac{w}{2} + \frac{w}{2\cos(\theta)})$.
+    # this puts the "edge" tip a distance $M = (w/2)\csc(\theta/2)$
+    # from the tip of the corner itself, on the line defined by the bisector of
+    # the corner angle. So the extra padding required is $M\sin(\phi)$, where
+    # $\phi$ is the incidence angle of the corner's bisector. Notice that in
+    # the limit ($\phi = \theta/2$) where the "corner" is flush with the bbox,
+    # this correctly simplifies to just $w/2$.
+    elif joinstyle == 'miter':
+        # matplotlib currently doesn't set the miterlimit...
+        _nominal_miter_limit = 10  # pdf and agg do this
+        if 1/np.sin(theta/2) > _nominal_miter_limit:
+            return _stroke_x_overflow(width, phi, theta, 'bevel', capstyle)
+        else:
+            return (width/2)*np.sin(phi)/np.sin(theta/2)
+    # a beveled edge is exactly the convex hull of its two composite lines with
+    # capstyle='butt'. So we just compute the individual lines' incidence
+    # angles and take the maximum of the two padding values
+    elif joinstyle == 'bevel':
+        phi1 = phi + theta/2
+        phi2 = phi - theta/2
+        return (width/2) * max(np.abs(np.cos(phi1)), np.abs(np.cos(phi2)))
+    # finally, _joinstyle = "round" is just _joinstyle = "bevel" but with
+    # a hemispherical cap. we could calculate this but for now no markers use
+    # it....except those with "no corner", in which case we can treat them the
+    # same as squares...
+    elif joinstyle == 'round':
+        return width/2  # hemispherical cap, so always same padding
+    else:
+        raise ValueError(f"Unknown joinstyle: {joinstyle}")
+
+
+def _pad_extents_stroked_vertex(extents, vinfo, markeredgewidth, joinstyle,
+                                capstyle):
+    """
+    Accumulator for building true extents from `.VertexInfo`s.
+
+    Parameters
+    ----------
+    extents : 4*[float]
+        The extents (xmin, ymin, xmax, ymax) of the `~.transforms.Bbox` of the
+        vertices. Modified in place so that the corner described by *vinfo*
```

Comment:
The type should be more explicit then; it has to a mutable reference of some sort to work (i.e., not tuples.)

Reply:
Is `(4,) array_like of float` better?

Reply:
Maybe, I'm not sure we have a convention for must-be-mutable, @timhoffm?